src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -144,17 +144,32 @@ const DateTimeBlock : React.FC<IProps> = ({
   }, [date, displayMinutes, setDateCallback]);
 
   const setDateMinutes = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
-    const minuteRegex = /^[0-5]{0,1}[0-9]{0,1}$/;
-    if (!minuteRegex.test(value)) {
-      return;
+    let newVal;
+    if(Number(value) > 59){
+      const minuteRegex = /^-?\d{1,2}$|^$/;
+      if (!minuteRegex.test(value)) {
+        return;
+      }
+      newVal = '00';
+    } else if(Number(value) < 0){
+      newVal = '59';
+    } else if(value.length === 1){
+      if(value === '0'){
+        newVal = ('00');
+      } else {
+        newVal = ('0' + value).slice(-2) ;
+      }
+    } else {
+      newVal = value.slice(-2);
```

Comment:
We have used slice method here to handle 2 digit value. 
For example

- If user enters "1" then the value appear in the input box is "01" and if in the same input box user enters 2nd digit let say "2" then the value appear in the input box is "12" after that no matter how many times user enter next digit it will show "12" value until user manually clear it.
- value.slice(-2) means slicing upto 2 digit